synthesizer/src/vm/verify.rs

```diff
@@ -233,8 +233,12 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
                 if let Some(fee) = fee {
                     // If the fee is required, then check that the base fee amount is satisfied.
                     if is_fee_required {
-                        // Compute the execution cost.
-                        let (cost, _) = execution_cost(&self.process().read(), execution)?;
+                        // Compute the execution cost based on the block height
+                        let block_height = &self.block_store().max_height().unwrap_or_default();
```

Comment:
```suggestion
                        let block_height = self.block_store().find_block_height_from_state_root(execution.global_state_root())?.unwrap_or_default();
```

This is a better alternative. `max_height()` is an extremely expensive call it is an O(N) call for the number of blocks.